x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/sections/action_connector_form/action_type_menu.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment, useContext } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCard,
+  EuiIcon,
+  EuiFlexGrid,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionType } from '../../../types';
+import { ActionsConnectorsContext } from '../../context/actions_connectors_context';
+import { useAppDependencies } from '../../app_dependencies';
+
+interface Props {
+  onActionTypeChange: (actionType: ActionType) => void;
+}
+
+export const ActionTypeMenu = ({ onActionTypeChange }: Props) => {
+  const { actionTypeRegistry } = useAppDependencies();
+  const { actionTypesIndex, setAddFlyoutVisibility } = useContext(ActionsConnectorsContext);
+  if (!actionTypesIndex) {
+    return null;
+  }
+
+  const actionTypes = Object.entries(actionTypesIndex)
+    .filter(([index]) => actionTypeRegistry.has(index))
+    .map(([index, actionType]) => {
+      const actionTypeModel = actionTypeRegistry.get(index);
+      return {
+        iconClass: actionTypeModel ? actionTypeModel.iconClass : '',
+        selectMessage: actionTypeModel ? actionTypeModel.selectMessage : '',
+        actionType,
+        name: actionType.name,
+        typeName: index.replace('.', ''),
+      };
+    });
+
+  const cardNodes = actionTypes.map((item, index): any => {
```

Comment:
Should filter actionTypes to ensure they are registered (UI & backend) and enabled on the backend.